src/protocol.rs

```diff
@@ -924,13 +932,11 @@ mod tests {
         let mut hasher = Sha256::new();
         hasher.update(b"some test message");
 
-        let signature = presign_outputs
+        let shares = presign_outputs
             .into_values()
             .map(|record| record.sign(hasher.clone()).unwrap())
-            .fold(SignatureShare::default(), |aggregator, share| {
-                aggregator.chain(share).unwrap()
-            })
-            .finish()?;
+            .collect::<Vec<_>>();
+        let signature = SignatureShare::into_signature(shares.into_iter())?;
```

Comment:
trivial suggestion: I think you don't actually have to `collect` the shares since this method takes an iterator.

Reply:
Good catch!